source/PlayerInfo.cpp

```diff
@@ -1356,7 +1366,21 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 	// Check if the incoming vector contains the same elements
 	if(std::is_permutation(ships.begin(), ships.end(), newOrder.begin()))
 	{
+		auto oldFirstShip = ships[0];
+		size_t pos = 0;
 		ships = newOrder;
+		// Check if the position of the flagship has changed.
+		if(ships[0] != oldFirstShip && Preferences::Has("Automatically unpark flagship"))
+		{
+			// Find the position of the previous flagship in the new vector.
+			pos = find(ships.begin(), ships.end(), oldFirstShip) - ships.begin();
+			// Check if the ship in the first position can be a flagship and is in the current system.
+			if(ships[0]->CanBeFlagship() && ships[0]->GetSystem() == system)
+			{
+				ships[0]->SetIsParked(false);
+				ships[pos]->SetIsParked(true);
+			}
+		}
```

Comment:
```suggestion
		if(newFirstShip != oldFirstShip && Preferences::Has("Automatically unpark flagship"))
		{
			// Find the position of the previous flagship in the new vector.
			pos = find(ships.begin(), ships.end(), oldFirstShip) - ships.begin();
			// Check if the ship in the first position can be a flagship and is in the current system.
			if(newFirstShip->CanBeFlagship() && newFirstShip->GetSystem() == system)
			{
				newFirstShip->SetIsParked(false);
				ships[pos]->SetIsParked(true);
			}
		}
```

Reply:
also consider using iterators instead of []

Reply:
Hey, thanks for the recommandation. As i am trying to learn with the Issue, I would like to know if it is improving the code when i change it to iterators? Or is it just coding style?

Reply:
iterators are more flexible and this for example:
pos = find(ships.begin(), ships.end(), oldFirstShip) - ships.begin();
already returns an iterator I think